Feature/io acthor heater element

DerStoecki opened the pull request:
AC Thor is a 3 phase heating element, with a maximum power of 1-3kW per phase.

This implementation reads from 3 digital inputs and sets the Power channel appropriately.

@clehne wenn du einmal kurz bevor ich das ins öffentliche Repo als PR stelle drüber gucken könntest wäre das super, ich habe noch 2 änderungen in der Nature vorgenommen (line break bei persistence prio) und ich hab einen basic test implementiert.

der PR hat den aktuellen develop branch daher sind noch "mehr" änderungen als nur der acthor drin im PR

DerStoecki committed:
Implements the AC Thor 3 phase heating Element

AC Thor is a 3 phase heating element, with a maximum power of 1-3kW per phase.

This implementation reads from 3 digital inputs and sets the Power channel appropriately.

Co-authored-by: Christian Lehne <51822163+clehne@users.noreply.github.com>

Sn0w3y commented on io.openems.edge.io.acthor/src/io/openems/edge/io/acthor/AcThorImpl.java:
```diff
@@ -0,0 +1,250 @@
+	}
+
+	private Integer calculatePower() {
+		Optional<Boolean> out1 = this.getOut1WriteChannel().getNextWriteValue(); // TODO Use getNextWriteValueAndReset
```
Maybe use `var `for more modern coding style

Sn0w3y commented:
Nur eine Frage meinerseits - so wie ich die Impl verstehe ist es so, dass der AC-Thor in deiner Version nicht von einem Controller gesteuert werden kann, oder?

Ich arbeite jetzt dann bald auch an einer anderen Implementierung eines Heizstabes (Smart mit Modbus) und Stefan meinte wir sollten das eventuell Abstahieren:

https://community.openems.io/t/steuerbarer-heizstab-fur-pv-uberschuss/2525/16?u=sn0w3y
https://community.openems.io/t/ac-thor-einbindung/2544/2?u=sn0w3y

Grüße !

clehne commented:
> Nur eine Frage meinerseits - so wie ich die Impl verstehe ist es so, dass der AC-Thor in deiner Version nicht von einem Controller gesteuert werden kann, oder?

Hallo @Sn0w3y, doch diese Implementierung kann direkt mit OpenEMS Bordmitteln genutzt werden. Habe dazu [hier](https://community.openems.io/t/ac-thor-einbindung/2544/5) ein paar Zeilen geschrieben. Danke für den Hinweis auf die beiden Community Posts.

Sn0w3y commented:
> > Nur eine Frage meinerseits - so wie ich die Impl verstehe ist es so, dass der AC-Thor in deiner Version nicht von einem Controller gesteuert werden kann, oder?
> 
> Hallo @Sn0w3y, doch diese Implementierung kann direkt mit OpenEMS Bordmitteln genutzt werden. Habe dazu [hier](https://community.openems.io/t/ac-thor-einbindung/2544/5) ein paar Zeilen geschrieben. Danke für den Hinweis auf die beiden Community Posts.

Alles klar, danke für den Hinweis ! :)

Aber es fehlt ja immer noch die PWM Steuerung, für die der Heizstab ja eigentlich auch benutzt werden sollte/kann, oder ?
Dadurch, dass er ja eben Stufenlos ist, sollte er ja auch so geregelt werden können, sonst kann man ja auch wieder gleich einen "dummen" Heizstab mit Relaisboard nehmen, oder ?

DerStoecki committed:
Refactored calculate Power

Instead of using the "if" cases and optionalBooleans
when getting the nextwrite value -> use orElse false directly
and use ternary operator
also i called nextWriteValueAndReset (was a TODO)
